authentik/sources/ldap/sync/base.py

```diff
@@ -140,62 +121,14 @@ def search_paginator(  # noqa: PLR0913
                 cookie = None
             yield self._connection.response
 
-    def build_user_properties(self, user_dn: str, **kwargs) -> dict[str, Any]:
-        """Build attributes for User object based on property mappings."""
-        props = self._build_object_properties(user_dn, **kwargs)
-        props.setdefault("path", self._source.get_user_path())
-        return props
-
-    def build_group_properties(self, group_dn: str, **kwargs) -> dict[str, Any]:
-        """Build attributes for Group object based on property mappings."""
-        return self._build_object_properties(group_dn, **kwargs)
-
-    def _build_object_properties(self, object_dn: str, **kwargs) -> dict[str, dict[Any, Any]]:
-        properties = {"attributes": {}}
-        try:
-            for value, mapping in self.mapper.iter_eval(
-                user=None,
-                request=None,
-                return_mapping=True,
-                ldap=kwargs,
-                dn=object_dn,
-                source=self._source,
-            ):
-                if isinstance(value, (bytes)):
-                    self._logger.warning("property mapping returned bytes", mapping=mapping)
-                    continue
-                object_field = mapping.object_field
-                if object_field.startswith("attributes."):
-                    # Because returning a list might desired, we can't
-                    # rely on flatten here. Instead, just save the result as-is
-                    set_path_in_dict(properties, object_field, value)
-                else:
-                    properties[object_field] = flatten(value)
-        except SkipObjectException as exc:
-            raise exc from exc
-        except PropertyMappingExpressionException as exc:
-            # Value error can be raised when assigning invalid data to an attribute
-            Event.new(
-                EventAction.CONFIGURATION_ERROR,
-                message=f"Failed to evaluate property-mapping {exception_to_string(exc)}",
-                mapping=exc.mapping,
-            ).save()
-            self._logger.warning("Mapping failed to evaluate", exc=exc, mapping=exc.mapping)
-            raise StopSync(exc, None, exc.mapping) from exc
```

Comment:
We need to decide what to do with this in the general case. The way I've written it currently is that if a property mapping fails, we just ignore it.

I think it's probably better to not do that, and fail whatever is currently happening (a sync or a source login flow). Thoughts?

Reply:
done, we now fail with `PropertyMappingExpressionException` if the expression fails to evaluate

The LDAP sync code then catches this and throws StopSync, to preserve existing behaviour.